### ISSUE TITLE
nix-store find wrong runtime gcroot when run containered nixos guest with systemd-nspawn

### ISSUE THREAD

**Issue by yuchangyuan** [NixOS/nix#2304]
When I run guest NixOS of same version with systemd-nspawn(**NOT** nixos-container) as container, nix-store on host OS will return wrong gcroots(show as "{memory:*}"), which are actual runtime gcroots of **guest OS**, which happens to be the same path on host OS.

**Comment by lheckemann** [comment]
Good catch! I haven't reproduced this issue myself, but it makes sense. I'm not sure how one would go about determining that they're actually distinct paths to fix this though.
